### PR TITLE
Clean up delete code

### PR DESCRIPTION
**Pull request by chbatey**
I found the nested flatmaps hard to follow

### CODE REVIEW

File: core/src/main/scala/akka/persistence/cassandra/journal/CassandraDeletion.scala [hunk 1]
```diff
@@ -50,68 +50,59 @@ private[akka] trait CassandraDeletion extends CassandraStatements {
...
+      highestDeletedSequenceNumber <- asyncHighestDeletedSequenceNumber(persistenceId)
+      lowestSequenceNr <- asyncReadLowestSequenceNr(persistenceId, 1L, highestDeletedSequenceNumber, Some(config.readConsistency), Some(readRetryPolicy))
+      highestSequenceNr <- asyncFindHighestSequenceNr(persistenceId, lowestSequenceNr, config.targetPartitionSize)
+      lowestPartition = partitionNr(lowestSequenceNr, config.targetPartitionSize)
+      toSeqNr = math.min(toSequenceNr, highestSequenceNr)
+      highestPartition = partitionNr(toSeqNr, config.targetPartitionSize) + 1 // may have been moved to the next partition
+      boundInsertDeletedTo = preparedInsertDeletedTo.map(_.bind(persistenceId, toSeqNr: JLong))
+      logicalDelete = boundInsertDeletedTo.flatMap(session.executeWrite)
+      _ = logicalDelete.foreach(_ => physicalDelete(lowestPartition, highestPartition, toSeqNr))
+      done <- logicalDelete.map(_ => Done)
```
Comment (johanandren) [inline core/src/main/scala/akka/persistence/cassandra/journal/CassandraDeletion.scala]: The last map+foreach inside of map still is a bit hard to understand (and a bit weird given that it side effects and then in parallel maps them to done), extract to a method or something perhaps?

Comment (chbatey) [reply core/src/main/scala/akka/persistence/cassandra/journal/CassandraDeletion.scala]: :+1:

Comment (chbatey) [reply core/src/main/scala/akka/persistence/cassandra/journal/CassandraDeletion.scala]: Yeah it is a bit mad how we have to delete given we have this "logical" delete with the metadata table and the real delete where the events might be in different partitions due to the way they were written :(

I've pushed a further refactor that makes this more clear (i hope)

Comment (t3hnar) [reply core/src/main/scala/akka/persistence/cassandra/journal/CassandraDeletion.scala]: I think it might get much cleaner after extracting event and metadata related queries into different classes.
